Tests/SentryTests/Integrations/SentryNetworkTrackerTests.swift

```diff
@@ -263,6 +257,96 @@ class SentryNetworkTrackerTests: XCTestCase {
         assertStatus(status: .undefined, state: .completed, response: URLResponse())
     }
     
+    func testResumeAfterCompleted_OnlyOneSpanCreated() {
```

Comment:
Did this fail before the large lock around was added?

Reply:
Or in other words: Any of the tests verifies the fix added?

Reply:
Yes, this verifies a fix. The test fails if I remove the large lock.